concourse/model/traits/image_scan.py

```diff
@@ -230,6 +258,10 @@ def clam_av(self):
         if self.raw['clam_av']:
             return ClamAVScanCfg(raw_dict=self.raw['clam_av'])
 
+    def compliance_db(self):
+        if self.raw['compliance_db']:
```

Comment:
this check will not do what you intend.

rather use `.get('compliance_db')` (or `if 'compliance_db' in self.raw).

since python3.8, you may use assignment expressions, which could be considered somewhat more elegant (not every python developer seems to agree, though):

```
if compliance_db_dict := self.raw.get('compliance_db', None'): # None added for explicitness - can be omitted
  return ComplianceDBCfg(raw_dict=compliance_db_dict)
else:
 return None # also: added for explicitness - can be omitted
```

also, consider adding a return-type-hint (-> typing.Optional[ComplianceDBCfg])

Reply:
also: since the attr is configured to be `required`, there would be no need to check for its existence. since we do not currently have a compliance-db-cfg everywhere, I recommend, to change the attr to optional..

Reply:
Wanted to keep consistency within `image_scan.py` as this is the current pattern (I've seen other traits use your mentioned pattern).
Will apply pattern and change attr to optional.